events/source.go

```diff
@@ -631,19 +630,29 @@ func ctxWithTimeout(ctx context.Context, timeout time.Duration) (context.Context
 	return context.WithTimeout(ctx, timeout)
 }
 
-// BlockClientFactoryAdapter adapts a Tendermint ClientFactory to a BlockClientFactory
-type BlockClientFactoryAdapter struct{ f ClientFactory }
+// BlockClientFactoryAdapter1 adapts a Tendermint ClientFactory to a BlockClientFactory
+type BlockClientFactoryAdapter1 struct{ Factory ClientFactory }
 
 // Create implements the BlockClientFactory interface
-func (a BlockClientFactoryAdapter) Create() (BlockClient, error) {
-	c, err := a.f.Create()
+func (a BlockClientFactoryAdapter1) Create() (BlockClient, error) {
+	c, err := a.Factory.Create()
 	return NewBlockClient(c), err
 }
 
-type blockResultsClientFactoryAdapter struct{ client BlockResultClient }
+// BlockClientFactoryAdapter2 adapts a Tendermint BlockClient to a BlockClientFactory
+type BlockClientFactoryAdapter2 struct{ Client BlockClient }
+
+// Create implements the BlockClientFactory interface
+func (a BlockClientFactoryAdapter2) Create() (BlockClient, error) {
```

Comment:
🤮

Reply:
if you have better names for these crappy adapters I'm all ears